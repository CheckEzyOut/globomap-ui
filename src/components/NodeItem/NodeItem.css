/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.node-item,
.node-item .graph-labels {
  box-sizing: border-box;
}

.node-item {
  position: relative;
  margin: 0 40px 30px 0;
  min-width: 120px;
  max-width: 200px;
  min-height: 80px;
  padding: 20px;
  cursor: pointer;
  background-color: white;
  border-radius: 5px;
  box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.2);
}

.node-item .node-info,
.node-item .node-graphs {
  /* padding: 10px 15px; */
}

.sticky {
  position: -webkit-sticky;
  position: sticky;
  top: 20px;
  z-index: 0;
}

.sticky:before,
.sticky:after {
  content: '';
  display: table;
}

.node-item .type {
  display: block;
  font-size: 14px;
  font-family: 'opensans-bold', Arial, Helvetica, sans-serif;
  color: #555;
  margin: 0 10px 6px 0;
  text-transform: uppercase;
}

.node-item .name {
  display: block;
  text-decoration: none;
  font-size: 14px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  margin-bottom: 6px;
}

.node-item.disabled {
  border-bottom-color: rgba(0, 0, 0, 0.1);
  background-color: rgba(157, 172, 169, 0.4);
}

.node-item.disabled .type {
  color: rgba(51, 51, 51, 0.4);
}

.node-item.disabled .name {
  color: rgba(255, 255, 255, 0.4);
}

.node-item.current {
  background-color: #fefefe;
}

.node-item.current .type,
.node-item.current .name {
  color: #2eb78e;
}

.node-item.current.this-node {
  /* border: 1px solid #587bff; */
  /* border-bottom-color: #0036ff; */
}

.node-item .sub-node-edges .edges-in,
.node-item .sub-node-edges .edges-out {
  position: absolute;
  left: -30px;
}

.node-item .sub-node-edges .edges-in {
  top: 2px;
}

.node-item .sub-node-edges .edges-out {
  bottom: 2px;
}

.node-item .close-node-btn {
  color: transparent;
  position: absolute;
  top: 5px;
  right: 2px;
  border: 0;
  background-color: transparent;
  font-size: 12px;
  z-index: 1;
}

.node-item:hover .close-node-btn {
  color: rgba(0, 0, 0, 0.2);
}

.node-item:hover .close-node-btn:hover {
  color: rgba(0, 0, 0, 0.9);
}
