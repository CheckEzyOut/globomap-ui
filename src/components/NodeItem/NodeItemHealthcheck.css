.node-item-hc {
  position: absolute;
  width: 15px;
  height: 15px;
  top: 0;
  right: 0;
  border-top-right-radius: 5px;
  overflow: hidden;
}

.node-item-hc .hc-result {
  position: absolute;
  width: 30px;
  height: 30px;
  top: -30px;
  right: -30px;
  border-radius: 50%;
  background-color: #ccc;
  transition: all 300ms ease-out;
}

.node-item-hc.active .hc-result {
  top: -15px;
  right: -15px;
}
