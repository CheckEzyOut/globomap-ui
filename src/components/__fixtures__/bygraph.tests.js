export const items = {
  "graph": "acl",
  "edges": [],
  "nodes": [
    {
      "_id": "tag_firewall/gproxy_LnMzLXNhLWVhc3QtMS5hbWF6b25hd3MuY29t",
      "id": "LnMzLXNhLWVhc3QtMS5hbWF6b25hd3MuY29t",
      "name": ".s3-sa-east-1.amazonaws.com",
      "provider": "gproxy",
      "timestamp": 1512025259,
      "properties": null,
      "properties_metadata": null,
      "type": "tag_firewall"
    }
  ],
  "subnodes": []
};

export const graphs = [
  {
    "name": "acl",
    "colorClass": "graph-color0",
    "enabled": false
  },
  {
    "name": "cloud_stack",
    "colorClass": "graph-color1",
    "enabled": false
  },
  {
    "name": "custeio",
    "colorClass": "graph-color2",
    "enabled": false
  },
  {
    "name": "dns",
    "colorClass": "graph-color3",
    "enabled": false
  },
  {
    "name": "galeb",
    "colorClass": "graph-color4",
    "enabled": false
  },
  {
    "name": "load_balancing",
    "colorClass": "graph-color5",
    "enabled": false
  },
  {
    "name": "networking_topology",
    "colorClass": "graph-color6",
    "enabled": false
  },
  {
    "name": "permission",
    "colorClass": "graph-color7",
    "enabled": false
  },
  {
    "name": "tsuru",
    "colorClass": "graph-color8",
    "enabled": false
  },
  {
    "name": "zabbix",
    "colorClass": "graph-color9",
    "enabled": false
  }
];
