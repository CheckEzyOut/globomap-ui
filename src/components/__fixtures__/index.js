export const node = {
  "_id": "comp_unit/globomap_5d9ed63b-1985-4d3a-98d4-8b0195ce1c53",
  "id": "5d9ed63b-1985-4d3a-98d4-8b0195ce1c53",
  "name": "cartolaapi-02-145989229527",
  "provider": "globomap",
  "timestamp": 1511441091,
  "properties": {
    "equipment_type": "Servidor Virtual",
    "host": "cmal18mp11lc03",
    "ips": [
      "10.132.67.64"
    ],
    "uuid": "5d9ed63b-1985-4d3a-98d4-8b0195ce1c53"
  },
  "properties_metadata": {
    "equipment_type": {
      "description": "Equipment Type"
    },
    "host": {
      "description": "Host name"
    },
    "ips": {
      "description": "IPs"
    },
    "uuid": {
      "description": "UUID"
    }
  },
  "type": "comp_unit",
  "edges": {
    "in": [
      {
        "_id": "access/aclapi_251406",
        "_from": "network/napi_v4_36200",
        "_to": "comp_unit/globomap_5d9ed63b-1985-4d3a-98d4-8b0195ce1c53",
        "id": "251406",
        "name": "10.130.7.224/27-10.132.67.64/32",
        "provider": "aclapi",
        "timestamp": 1512019453,
        "properties": {
          "protocol": "tcp",
          "destination": "10.132.67.64/32",
          "source": "10.130.7.224/27",
          "action": "permit"
        },
        "properties_metadata": {
          "protocol": {
            "description": "Protocol of access"
          },
          "destination": {
            "description": "Destination address"
          },
          "source": {
            "description": "Source address"
          },
          "action": {
            "description": "Access action"
          }
        },
        "type": "access",
        "dir": "in"
      }
    ],
    "out": [],
    "graph": "acl"
  },
  "uuid": "2dc992f7-f9fd-432a-a9d6-b0484d4861b3",
  "items": [],
  "exist": true
};

export const nodes = [
  {
    "_id": "tag_firewall/gproxy_LmdpdGh1Yi1jbG91ZC5zMy5hbWF6b25hd3MuY29t",
    "id": "LmdpdGh1Yi1jbG91ZC5zMy5hbWF6b25hd3MuY29t",
    "name": ".github-cloud.s3.amazonaws.com",
    "provider": "gproxy",
    "timestamp": 1511247666,
    "properties": null,
    "properties_metadata": null,
    "type": "tag_firewall"
  },{
    "_id": "tag_firewall/gproxy_LnMzLXNhLWVhc3QtMS5hbWF6b25hd3MuY29t",
    "id": "LnMzLXNhLWVhc3QtMS5hbWF6b25hd3MuY29t",
    "name": ".s3-sa-east-1.amazonaws.com",
    "provider": "gproxy",
    "timestamp": 1511247655,
    "properties": null,
    "properties_metadata": null,
    "type": "tag_firewall"
  }
];
