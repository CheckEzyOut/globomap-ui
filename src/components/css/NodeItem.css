.node-item,
.node-item .graph-labels {
  box-sizing: border-box;
}

.node-item {
  position: relative;
  margin: 0 10px 10px 0;
  border-radius: 2px;
  min-width: 120px;
  max-width: 200px;
  cursor: pointer;
  background-color: #9daca9;
  border: 1px solid transparent;
  border-bottom: 1px solid rgba(0, 0, 0, 0.3);
}

.node-item .node-info,
.node-item .node-graphs {
  padding: 10px 15px;
}

.node-item .type {
  display: block;
  font-size: 10px;
  margin-bottom: 6px;
  text-transform: uppercase;
}

.node-item .name {
  display: block;
  color: white;
  text-decoration: none;
  font-size: 14px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.node-item.current {
  background-color: #7f8c89;
}

.node-item.current.this-node {
  border: 1px solid #587bff;
  border-bottom-color: #0036ff;
  box-shadow: 0 0 0 3px #6fb5f1;
}

.node-item .sub-node-edges .edges-in,
.node-item .sub-node-edges .edges-out {
  position: absolute;
  left: -16px;
}

.node-item .sub-node-edges .edges-in {
  top: 2px;
}

.node-item .sub-node-edges .edges-out {
  bottom: 2px;
}
