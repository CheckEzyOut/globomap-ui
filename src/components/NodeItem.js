import React, { Component } from 'react';
import './css/NodeItem.css';

class NodeItem extends Component {
  constructor(props) {
    super(props);
    this.onItemSelect = this.onItemSelect.bind(this);
  }

  render() {
    let node = this.props.node,
        { name, current } = node;

    return (
      <div className={"node-item " + (current ? " current" : "")}
           onClick={this.onItemSelect}>
        <a href={"#"+ name} className="node-name">{name}</a>
      </div>
    );
  }

  onItemSelect(event) {
    event.stopPropagation();
    this.props.setCurrent(this.props.node._id);
  }
}

export default NodeItem;
