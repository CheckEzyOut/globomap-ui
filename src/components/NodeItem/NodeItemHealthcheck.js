/*
Copyright 2019 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import axios from 'axios';
import React, { Component } from 'react';
import { host } from '../../config'
import './NodeItemHealthcheck.css';

export class NodeItemHealthcheck extends Component {

  constructor(props) {
    super(props);

    // plugin name that this component is enabled
    this.pluginName = 'healthcheck';

    this.state = {
      result: null
    };

    this.isEnabled = this.isEnabled.bind(this);
  }

  isEnabled() {
    const plugins = this.props.plugins;
    for (let i=0, l=plugins.length; i<l; i++) {
      if (plugins[i].name === this.pluginName) {
        return true;
      }
    }
    return false;
  }

  getHealthcheckStatus() {
    const plugin = this.props.plugins.filter(p => {
      return p.name === this.pluginName;
    })[0];

    if (plugin) {
      // axios.post(`${host}/api/plugins/${this.pluginName}`, { params: params });
      console.log('check status');
    }
  }

  async componentDidUpdate() {
    this.getHealthcheckStatus();
  }

  render() {
    if (!this.isEnabled()) {
      return null;
    }

    return (
      <div className="node-item-hc active">
        <div className="hc-result"></div>
      </div>
    );
  }

}

export default NodeItemHealthcheck;
