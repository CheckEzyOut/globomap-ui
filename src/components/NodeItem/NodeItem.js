/*
Copyright 2019 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import _ from "lodash";
import tippy from 'tippy.js';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import Stickyfill from 'stickyfill';
import {
  traversalSearch,
  resetSubNodes,
  setCurrentNode } from '../../redux/modules/nodes';
import {
  removeStageNode,
  setStageNodes } from '../../redux/modules/stage';
import { NodeEdges } from '../';
import NodeItemHealthcheck from './NodeItemHealthcheck';
import './NodeItem.css';

export class NodeItem extends Component {

  constructor(props) {
    super(props);

    this.state = {
      pluginsFiltered: false,
      pluginsLoaded: false,
      nodePlugins: []
    };

    this.ref = React.createRef();

    this.onItemSelect = this.onItemSelect.bind(this);
    this.onSelfRemove = this.onSelfRemove.bind(this);
    this.filterPlugins = this.filterPlugins.bind(this);
    this.stickyfill = Stickyfill();
  }

  onItemSelect(event) {
    if (!this.props.currentNode) {
      this.props.traversalSearch({ node: this.props.node });
      this.props.setCurrentNode(this.props.node);
      return;
    }

    if (this.props.currentNode._id !== this.props.node._id) {
      this.props.traversalSearch({ node: this.props.node });
    }

    this.props.setCurrentNode(this.props.node);
  }

  onNodeInfo(event) {
    event.stopPropagation();
    this.props.onShowNodeInfo();
  }

  onSelfRemove(event) {
    event.stopPropagation();
    this.props.removeStageNode(this.props.node);

    if (this.props.currentNode === null) {
      return;
    }

    if (this.props.currentNode.uuid === this.props.node.uuid) {
      this.props.resetSubNodes();
    }
  }

  filterPlugins() {
    if (this.state.pluginsFiltered) {
      return;
    }

    const plugins = this.props.plugins;
    if (!plugins) {
      return;
    }

    let nodePlugins = [];
    for (let i=0, l=plugins.length; i<l; i++) {
      const plugin = plugins[i];
      if (plugin.types.includes(this.props.node.type)) {
        nodePlugins.push(plugin);
      }
    }

    this.setState({
      pluginsFiltered: true,
      nodePlugins: nodePlugins
    });
  }

  addPluginsObserver() {
    const observer = new IntersectionObserver(
      ([entry]) => {
        if (entry.intersectionRatio === 1 && !this.state.pluginsLoaded) {
          this.setState({ pluginsLoaded: true });
        }
      }, { root: null, rootMargin: '0px', threshold: 1.0 }
    )

    if (this.ref.current) {
      observer.observe(this.ref.current)
    }
  }

  componentDidMount() {
    let element = document.getElementsByClassName('sticky');
    this.stickyfill.add(element);
    tippy('.btn-with-tip', { arrow: true, animation: "fade" });

    this.addPluginsObserver()
    this.filterPlugins();
  }

  componentDidUmount() {
    this.setState({ pluginsLoaded: false });
  }

  render() {
    let { _id, name, type, edges, uuid, id } = this.props.node;
    let cNode = this.props.currentNode,
        current = cNode && _id === cNode._id ? ' current' : '',
        thisnode = cNode && uuid === cNode.uuid ? ' this-node' : '',
        exist = this.props.node.exist === undefined ? true : this.props.node.exist,
        disabled = !exist ? ' disabled' : '';

    let nodeType = type;
    if (this.props.namedCollections !== undefined
        && this.props.namedCollections[type] !== undefined) {
      nodeType = this.props.namedCollections[type].alias;
    }

    return (
      <div key={this.props.node.id}
           className={'node-item' + disabled + current + thisnode}
           onClick={exist && _.debounce(this.onItemSelect, 100, true)}>

        <div className="node-info sticky">
          <span className="type">{nodeType}</span>
          <span ref={this.ref} className="name">{name}</span>
          {this.props.hasId && <span>{id}</span>}
        </div>

        <NodeEdges edges={edges} position={'left'} />

        <div className="node-item-tools">
          <button className="btn-with-tip show-node-info-btn"
                  onClick={e => this.onNodeInfo(e)}
                  data-tippy-content="Mostrar Propriedades">
            <i className="fa fa-info-circle"></i>
          </button>
          {!this.props.node.root &&
            <button className="btn-with-tip close-node-btn"
                    onClick={this.onSelfRemove}
                    data-tippy-content="Remover este item">
              <i className="fa fa-times-circle"></i>
            </button>}
        </div>

        {this.state.pluginsLoaded &&
          <NodeItemHealthcheck  node={this.props.node}
                                plugins={this.state.nodePlugins} />}
      </div>
    );
  }

}

function mapStateToProps(state) {
  return {
    currentNode: state.nodes.currentNode,
    stageNodes: state.stage.stageNodes,
    hasId: state.app.hasId,
    namedCollections: state.app.namedCollections,
    plugins: state.plugins.plugins
  };
}

export default connect(
  mapStateToProps,
  {
    setCurrentNode,
    removeStageNode,
    resetSubNodes,
    setStageNodes,
    traversalSearch
  }
)(NodeItem);
