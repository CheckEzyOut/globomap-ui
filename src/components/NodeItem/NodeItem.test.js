import React from 'react';
import { shallow } from 'enzyme';
import { NodeItem } from './NodeItem';
import { node } from '../__fixtures__';

describe("NodeItem", function() {
  it('should render NodeItem component', () => {
    const component = shallow(<NodeItem node={node} />);
    expect(component).toMatchSnapshot();
  });
});
