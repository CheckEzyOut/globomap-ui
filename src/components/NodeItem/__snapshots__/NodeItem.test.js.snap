// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeItem should render NodeItem component 1`] = `
<div
  className="node-item"
  key="5d9ed63b-1985-4d3a-98d4-8b0195ce1c53"
  onClick={[Function]}
>
  <div
    className="node-info sticky"
  >
    <span
      className="type"
    >
      comp_unit
    </span>
    <span
      className="name"
    >
      cartolaapi-02-145989229527
    </span>
  </div>
  <Connect(NodeEdges)
    edges={
      Object {
        "graph": "acl",
        "in": Array [
          Object {
            "_from": "network/napi_v4_36200",
            "_id": "access/aclapi_251406",
            "_to": "comp_unit/globomap_5d9ed63b-1985-4d3a-98d4-8b0195ce1c53",
            "dir": "in",
            "id": "251406",
            "name": "10.130.7.224/27-10.132.67.64/32",
            "properties": Object {
              "action": "permit",
              "destination": "10.132.67.64/32",
              "protocol": "tcp",
              "source": "10.130.7.224/27",
            },
            "properties_metadata": Object {
              "action": Object {
                "description": "Access action",
              },
              "destination": Object {
                "description": "Destination address",
              },
              "protocol": Object {
                "description": "Protocol of access",
              },
              "source": Object {
                "description": "Source address",
              },
            },
            "provider": "aclapi",
            "timestamp": 1512019453,
            "type": "access",
          },
        ],
        "out": Array [],
      }
    }
    position="left"
  />
  <div
    className="node-item-tools"
  >
    <button
      className="btn-with-tip show-node-info-btn"
      data-tippy-content="Show Properties"
      onClick={[Function]}
    >
      <i
        className="fa fa-info-circle"
      />
    </button>
    <button
      className="btn-with-tip close-node-btn"
      data-tippy-content="Remove this node"
      onClick={[Function]}
    >
      <i
        className="fa fa-times-circle"
      />
    </button>
  </div>
</div>
`;
